tetris.js

```diff
@@ -1,185 +1,207 @@
 module.exports = {
 	//тетрамино массив из 4 координат [{x: 1, y: 2}, {x: 2, y: 2}, {x: 3, y: 2}, {x: 4, y: 2}]
 	// field это массив двухмерный булин значений, если true в ней есть блок, false блока нет
-	rotateLine: function(field, tetromino){
+	rotateLine: function (field, tetromino) {
 		let lengthField_X = field[0].length;
 		let lengthField_Y = field.length;
 		let turningPoint = [];
-		let arrCoordinates_X  = [];
-		let arrCoordinates_Y  = [];
+		let arrCoordinates_X = [];
+		let arrCoordinates_Y = [];
 		let a = -1;
 		let localTetromino = []
-		
-		tetromino.forEach(function(elem){
-			localTetromino.push(Object.assign({},elem));
+
+
+		tetromino.forEach(function (coordinates) {
+			localTetromino.push(Object.assign({}, coordinates));
 		})
-		tetromino.forEach(function(elem){
-			turningPoint.push(Object.assign({},elem));
+		tetromino.forEach(function (coordinates) {
+			turningPoint.push(Object.assign({}, coordinates));
 		})
-		
-		for(let elem of turningPoint){
+
+		for (let elem of turningPoint) {
 			arrCoordinates_X.push(elem.x);
 			arrCoordinates_Y.push(elem.y);
 		}
-		
-		if((arrCoordinates_Y[0] == arrCoordinates_Y[1]) && (arrCoordinates_Y[2] == arrCoordinates_Y[3]) && (arrCoordinates_Y[1] == arrCoordinates_Y[2])){
-			
-			for(let i of turningPoint){
+
+		function checkThatTheFieldIsFree(arrayOfCoordinates) {
+			if (allItemsOnTheField(arrayOfCoordinates) === false) {
+				return false
+			}
+			for (let coordinate of arrayOfCoordinates) {
+				if (field[coordinate.y][coordinate.x] === true) {
+					return false
+				}
+			}
+			return true;
+		}
+
+		function allItemsOnTheField(arrayOfCoordinates) {
+			for (let coordinate of arrayOfCoordinates) {
+				if (
+					(coordinate.x < 0 || coordinate.y < 0) ||
+					(coordinate.x > (lengthField_X - 1)) ||
+					(coordinate.y > (lengthField_Y - 1))
+				) {
+					return false;
+				}
+			}
+			return true
+		}
+
+		if ((arrCoordinates_Y[0] === arrCoordinates_Y[1]) && (arrCoordinates_Y[2] === arrCoordinates_Y[3]) && (arrCoordinates_Y[1] === arrCoordinates_Y[2])) {
+
+			for (let i of turningPoint) {
 				i.x = arrCoordinates_X[1];
 				i.y = arrCoordinates_Y[0] + a;
 				a++;
 			}
-			let newPosition = allItemsOnTheField(turningPoint)
-			if(checkThatTheFieldIsFree(newPosition)){
-				return localTetromino
+
+			 if(checkThatTheFieldIsFree(turningPoint)){
+				return turningPoint
```

Comment:
I don't like the name of this variable - point means it's one coordinate, but here we have the collection of the coordinates. And I don't see anyone turning around this collection, so it's not really turning, right?